### ISSUE TITLE
Issues installing with more than 1 SD Cards

### ISSUE THREAD

**Issue by dragoonDorise** [dragoonDorise/EmuDeck#8]
Test what happens when you use different cards and thumb drives

**Comment by Godsbane** [comment]
I dug into this a bit anyways...
I have a little multiport dongle with an embedded sd card reader.

When a card is inserted into the reader, it creates two new devices: /dev/sda and /dev/sdb
It also creates partitions for the sd card as /dev/sdb1 etc

when i then inserted a USB stick it created a new device for that as /dev/sdc and the associated partitions as /dev/sdc1, /dev/sdc2 etc

This all could be quirks of my particular usb-c dongle, but i don't have another to test with at the moment.

long story short - i don't believe it's feasible to support drives past the internal and the built in sd card reader at this point.

**Comment by bmfrosty** [comment]
You would need to do some scripting around device discovery.  Would be a pain to do in bash.

**Comment by Godsbane** [comment]
The secondary drives don't even mount automatically. You need to explicitly mount them, which i think makes the experience worse - and I don't see a way to mount them in steamos game mode.

**Comment by Godsbane** [comment]
So if you decide you want to pursue this with the caveat that the user has to mount the filesystem:

```
deck@steamdeck 1000)$ mount_arr=( $(findmnt --raw --evaluate --output=target -t ext4 | grep /run/media ))
deck@steamdeck 1000)$ echo ${mount_arr[0]}
/run/media/external
(deck@steamdeck 1000)$ echo ${mount_arr[1]}
/run/media/deck/28bf5693-635c-4b79-98cd-0f1c4b879c28
(deck@steamdeck 1000)$ echo ${mount_arr[@]}
/run/media/external /run/media/deck/28bf5693-635c-4b79-98cd-0f1c4b879c28
(deck@steamdeck 1000)$ echo ${#mount_arr[@]}
2

proofing out the data:

(deck@steamdeck 1000)$ findmnt --raw --evaluate -T ${mount_arr[1]}
TARGET SOURCE FSTYPE OPTIONS
/run/media/deck/28bf5693-635c-4b79-98cd-0f1c4b879c28 /dev/sdb1 ext4 rw,nosuid,nodev,relatime,errors=remount-ro

(deck@steamdeck 1000)$ findmnt --raw --evaluate -T ${mount_arr[0]}
TARGET SOURCE FSTYPE OPTIONS
/run/media/external /dev/mmcblk0p1 ext4 rw,noatime
(deck@steamdeck 1000)$ 

```

1. this should be useful so you can find user mounted ext4 filesystems and then iterate through them. 
2. present the user with the list of devices / paths.
3. set the user selected filesystem as the base path
4. continue the rest of the script.

**Comment by Godsbane** [comment]
here's a quick demonstration of this working. 

```

mount_arr=( $(findmnt --raw --evaluate --output=target -t ext4 | grep /run/media ))
for (( i=0; i<${#mount_arr[@]}; i++ ))
do findmnt -n --raw --evaluate --output=target -T ${mount_arr[$i]}
done
```

```
/run/media/external
/run/media/deck/28bf5693-635c-4b79-98cd-0f1c4b879c28
/run/media/deck/test64
```

zenity --entry --title "Select Device" --text "~/" --text "What Device would you like to use?" "${mount_arr[@]}"

![image](https://user-images.githubusercontent.com/1683513/163692556-b53048fc-48a8-4e42-aec4-b17fb9aafd27.png)

**Comment by dragoonDorise** [comment]
I've found that you can use multiple SD cards as long as they are all EXT4, as the run/media folder doesn't change.

The issue comes when using non EXT4 cards, I've also found that when using non EXT4 cards the script can't create symlinks as is if it didn't have enough permissions, can anyone try if they have the same problems?

**Comment by Godsbane** [comment]
for btrfs sd cards, the deck user doesn't have permissions. it's owned by root:root.

**Comment by Witherking25** [comment]
It shouldn’t be owned by root:root. How are you formatting it?

**Comment by Godsbane** [comment]
I just used kde partition manager.

**Comment by Witherking25** [comment]
Otherwise it will get mounted as root:root

**Comment by Godsbane** [comment]
No I understand that, however, this is explaining default behavior.

As of now, steamos does not support btrfs _out of the box._ I was more explaining to to @dragoonDorise one of the issues we encountered when testing it with stock settings, with btrfs.

We absolutely can support users who have done mods to their devices, and it shouldn't have any bearing on what the script does.

Whether the user is using btrfs or ext4 shouldn't and doesn't matter to us. The key is does steamos handle it, and does it support symlinks.

I'd like to add that all this _other_ filesystem talk is muddying the waters of this specific issue, and a separate issue should be opened for that discussion.

**Comment by Witherking25** [comment]
btrfs totally supports symlinks

**Comment by Godsbane** [comment]
fixed by adding a custom user choice to the installation
Fixed by #198
